melodies_monet/util/tools.py

```diff
@@ -498,13 +498,90 @@ def convert_std_to_amb_bc(ds,convert_vars=[],temp_var=None,pres_var=None):
     #So I just need to convert the obs from std to amb.
     Losch = 2.69e25 # loschmidt's number
     #1013 mb, 273 K (101300 Pa)
-    std_bc = 101300.*6.02214e23/(8.314472*273.)
+    std_bc = 101300.*N_A/(R*273.)
     #use pressure_obs now, which is in pa
-    Airnum = ds[pres_var]*6.02214e23/(8.314472*ds[temp_var])
+    Airnum = ds[pres_var]*N_A/(R*ds[temp_var])
     
     # amb to std = Losch / Airnum
     convert_std_to_amb_bc = Airnum/std_bc
     
     for var in convert_vars:
         ds[var] = ds[var]*convert_std_to_amb_bc
 
+
+def calc_partialcolumn(modobj, var="NO2"):
+    """Calculates the partial column of a species from its concentration.
+
+    Parameters
+    ----------
+    modobj : xr.Dataset
+        Model data
+    var : str
+        variable to calculate the partial column from
+
+    Returns
+    -------
+    xr.DataArray
+        DataArray containing the partial column of the species.
+    """
+    ppbv2molmol = 1e-9
+    m2_to_cm2 = 1e4
+    fac_units = ppbv2molmol * N_A / m2_to_cm2
+    partial_col = (
+        modobj[var]
+        * modobj["pres_pa_mid"]
+        * modobj["dz_m"]
+        * fac_units
+        / (R * modobj["temperature_k"])
+    )
+    partial_col.attrs = {"units": "molecules/cm2", "long_name": f"{var} partial column"}
+    return partial_col
+
+
+def calc_totalcolumn(modobj, var="NO2"):
+    """Calculates the total column of a species from its concentration.
+
+    Parameters
+    ----------
+    modobj : xr.Dataset
+        Model data
+    var : str
+        variable to calculate the total column from
+
+    Returns
+    -------
+    xr.DataArray
+        DataArray containing the total column of the species.
+    """
+    data = calc_partialcolumn(modobj, var)
+    try:
+        data = data.where(modobj['pres_pa_mid'] <= modobj['surfpres_pa'])
+    except KeyError:
+        pass
+    total_col = data.sum(dim='z', keep_attrs=True)
+    total_col.attrs = {"units": "molecules/cm2", "long_name": f"{var} total column"}
+    return total_col
+
+
+def calc_geolocaltime(modobj):
+    """Calculates the geographic local time based on the longitude.
+
+    Parameters
+    ----------
+    modobj : xr.Dataset
+        Model data
+
+    Returns
+    -------
+    xr.DataArray
+        DataArray containing the local time based on longitude.
+    """
+    # Make sure that lon is in the range [-180, 180]
+    # This should be guaranteed by the reader, and it isn't needed,
+    # but it is very cheap to redo and should make us be safer.
+
+    hrs2ms = 3600000
```

Comment:
```suggestion
    hrs2ms = 3600_000
```
Might help with clarity and with checking the number of zeros at a glance ([new in Python 3.6](https://docs.python.org/3/whatsnew/3.6.html#pep-515-underscores-in-numeric-literals), if that's relevant to support considerations)